src/test/java/health/ere/ps/service/dgc/DigitalGreenCertificateServiceTest.java

```diff
@@ -137,4 +135,48 @@ void issueVaccinationCertificate() {
         assertEquals(response, digitalGreenCertificateService.issueVaccinationCertificatePdf(fn, gn, dob, id, tg, vp, mp,
                 ma, dn, sd, dt));
     }
+
+    @Test
+    void issueRecoveryCertificatePdf() {
+        String fn = "testFn";
+
+        String gn = "testGn";
+
+        LocalDate dob = LocalDate.of(2000, 1, 1);
+
+        String id = "testId";
+
+        String tg = "testTg";
+
+        LocalDate fr = LocalDate.of(2021, 5, 31);
+
+        String is = "testIs";
+
+        LocalDate df = LocalDate.of(2021, 7, 1);
+
+        LocalDate du = LocalDate.of(2022, 7, 1);
+
+        byte[] bytes = new byte[]{56, 67, 78};
+
+        RecoveryCertificateRequest recoveryCertificateRequest = new RecoveryCertificateRequest();
+
+        recoveryCertificateRequest.setNam(new PersonName(fn, gn));
+        recoveryCertificateRequest.setDob(dob);
+
+        RecoveryEntry r = new RecoveryEntry();
+
+        r.setId(id);
+        r.setTg(tg);
+        r.setFr(fr);
+        r.setIs(is);
+        r.setDf(df);
+        r.setDu(du);
+
+        recoveryCertificateRequest.setR(Collections.singletonList(r));
+
+        doReturn(bytes).when(digitalGreenCertificateService).issuePdf(recoveryCertificateRequest);
```

Comment:
```suggestion
        // doReturn because of the null check in issuePdf
        doReturn(bytes).when(digitalGreenCertificateService).issuePdf(recoveryCertificateRequest);
```